front_end/Matching/ResponseList.js

```diff
@@ -0,0 +1,189 @@
+import React, {Component} from 'react';
+import {
+    ScrollView,
+    StyleSheet, 
+    Text,
+    TouchableOpacity,
+    View,
+    TextInput
+} from 'react-native';
+
+import Icon from 'react-native-vector-icons/Ionicons';
+import ReadMore from 'react-native-read-more-text';
+
+class Response extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            response: props.response,
+            username: props.username,
+            response_id: props.response_id,
+            reply: "",
+            disabled: true,
+            expand: false,
+            //nav: props.nav,
+            answered: false,
+            sent: false
+        };
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSend = this.handleSend.bind(this);
+    }
+
+    handleChange(value){
+        this.setState({reply: value});
+        if(value === ""){
+            this.setState({disabled: true});
+        } else{
+            this.setState({disabled: false});
+        }
```

Comment:
1) Never set state more than once in a function. Setting state is a very expensive operation as it rerenders everything.

2) This can be a one-liner: 
```
this.setState({reply: value, disabled: (value === "")});
```
Thus, is it even worth it to have a function, rather than just to keep it inline in the render function? I'll leave that up to you.